compiler/next/include/chpl/uast/Domain.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_DOMAIN_H
+#define CHPL_UAST_DOMAIN_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Expression.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a domain expression. For example:
+
+  \rst
+  .. code-block:: chapel
+
+      // Example 1:
+      var d = { 'foo', 'bar', 'baz' };
```

Comment:
I think this also supports `{1..10}` because the expression inside will be a range?

Reply:
Yes.